docs/OperatorBestPractices.md

```diff
@@ -1,26 +1,59 @@
-# Operator Best Practices
+# Backing Service Best Practices
 
 ## Introduction
 
 The goals of the Service Binding Operator is to make it easier for
 applications developers to bind applications with needed backing
-services, without having to perform manual configuration of secrets,
-configmaps, etc. and to assist operator providers in promoting and
+services, without having to perform manual configuration of `secrets`,
+`configmaps`, etc. and to assist operator providers in promoting and
 expanding the adoption of their operators.
 
-When a ServiceBindingRequest is created the Service Binding Operator
+When a `ServiceBindingRequest` is created the Service Binding Operator
 collects binding information and shares it with application. The
 Binding Service Operator's controller injects the binding information
-into the application's "DeploymentConfig", "Deployment" or “Replicaset"
-as environment variables via an intermediate Secret called "binding-request".
-The binding also works with Knative services as it works with any "deployment"
-controller which has the podspec defined in the its jsonpath as
+into the application's `DeploymentConfig`, `Deployment` or `Replicaset`
+as environment variables via an intermediate Secret.
+The binding also works with Knative `Services` as it works with any API which has the podspec defined in the its jsonpath as
 "spec.template.spec.containers".
 
 This document provides "best practices" guidelines for the development of
 Operators that manage backing services to be bound together with applications
 by the Service Binding Operator.
 
+## Making a Kubernetes Resource Bindable
+
+In order to make backing service bindable, the Kubernetes resource representing the same needs to be annotated as a means to express what is "interesting" for applications.
+
+Example, in an `Ingress` or `Route` resource representing a backing service, the `host` and `port` among other things would be "interesting" for applications for connecting to the backing service.
+
+To do so, the Kubernetes resource may be annotated to meaningfully convey what is "interesting" for binding.
+
+``` yaml
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  name: example
+  namespace: service-binding-demo
+  annotations:
+    openshift.io/host.generated: 'true'
+    servicebindingoperator.redhat.io/spec.host: 'binding:env:attribute' #annotate here.
```

Comment:
it would be great if we could have a doc section about all possible values an annotation can have.

Reply:
I would like to re-organize the doc to include the new annotations' format anyway in a separate doc since it might be too much information without linking out. I plan to bring in this

https://github.com/k8s-service-bindings/spec#annotation-examples

In most non-CR resources,

```
 servicebindingoperator.redhat.io/spec.SOMETHING: 'binding:env:attribute'
```
is that'll work.

Reply:
Also, we gotta document use of Secrets/Configmaps which I haven't tried out, but I heard works.